etcd/etcd.go

```diff
@@ -167,6 +170,15 @@ func (c *Client) tryEndpoints(mreq func(string) (*http.Request, error)) (*http.R
 			break
 		}
 
+		if isTimeout(err) {
+			break
+		} else if uerr, ok := err.(*url.Error); ok {
+			if isTimeout(uerr.Err) {
+				err = uerr.Err
+				break
+			}
+		}
+
```

Comment:
yes, on purpose. the multiple endpoints are tried for the cases of outage or other connection problems. the timeout is not part of these retry causing errors, but it is used to identify the case when no new changes occured for a certain time. Or we can say so, too, that the timeout is not a remote problem, it is enforced on the client side, so it is independent of the health state of the remote endpoint. On a one step higher level, we use the timeouts in this client, to identify batches of updates, because otherwise etcd would notify skipper for each route change one by one.